drivers/FileHandle.h

```diff
@@ -100,43 +124,20 @@ class FileHandle {
      *    0 on success or un-needed,
      *   -1 on error
      */
-    virtual int fsync() = 0;
-
-    virtual off_t flen() {
-        lock();
-        /* remember our current position */
-        off_t pos = lseek(0, SEEK_CUR);
-        if(pos == -1) {
-            unlock();
-            return -1;
-        }
-        /* seek to the end to get the file length */
-        off_t res = lseek(0, SEEK_END);
-        /* return to our old position */
-        lseek(pos, SEEK_SET);
-        unlock();
-        return res;
-    }
-
-    virtual ~FileHandle() {};
-
-protected:
-
-    /** Acquire exclusive access to this object.
-     */
-    virtual void lock() {
-        // Stub
-    }
+    MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileLike::sync")
+    virtual int fsync() { return sync(); }
```

Comment:
I see you've removed lock - I'm going to need to think a bit about whether this is right or wrong.

Its semantics certainly weren't clear to me, so maybe it should go? I guess it's not part of the abstract API. All external calls should not need to call it, right? Okay, makes sense.

I think some lock might be necessary for the poll mechanism, but maybe that can go into a separate bit of machinery (PollableFileHandle?), like FileBase's lock. (Possibly protected?)